bb8/src/api.rs

```diff
@@ -119,14 +119,18 @@ impl<M: ManageConnection> Builder<M> {
     /// Constructs a new `Builder`.
     ///
     /// Parameters are initialized with their default values.
-    pub fn new() -> Builder<M> {
-        Default::default()
+    pub fn new() -> Self {
+        Builder::default()
     }
 
     /// Sets the maximum number of connections managed by the pool.
     ///
     /// Defaults to 10.
-    pub fn max_size(mut self, max_size: u32) -> Builder<M> {
+    ///
+    /// # Panics
+    /// Will panic if `max_size` is 0
```

Comment:
Style nit: can you add an empty line after `# Panics` and a period at the end of the following sentence, here and for the other affected cases?